Adding PMML changes to Kogito microservices doc (#61)
hmanwani-rh/kie-docs: doc-content/enterprise-only/getting-started-kogito/con-rhpam-kogito-operator-with-prometheus-and-grafana.adoc

[id="con-rhpam-kogito-operator-with-prometheus-and-grafana_{context}"]
= Red Hat Process Automation Manager {KOGITO} Operator interaction with Prometheus and Grafana

{KOGITO} in {PRODUCT} provides a `monitoring-prometheus-addon` add-on that enables Prometheus metrics monitoring for {KOGITO} microservices and generates Grafana dashboards that consume the default metrics exported by the add-on. The RHPAM {KOGITO} Operator uses the  https://github.com/coreos/prometheus-operator[Prometheus Operator] to expose the metrics from your project for Prometheus to scrape. Due to this dependency, the Prometheus Operator must be installed in the same namespace as your project.

When you deploy a {KOGITO} microservice that uses the `monitoring-prometheus-addon` add-on and the Prometheus Operator is installed, the Red Hat Process Automation Manager {KOGITO} Operator creates a `ServiceMonitor` custom resource to expose the metrics for Prometheus, as shown in the following example:

.Example `ServiceMonitor` resource for Prometheus
[source,yaml]
----
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  labels:

>     app: onboarding-service

Reviewer: I would suggest to change the name of the example. Onboarding-service example utilizes also processes and processes are not planned to work in Kogito in RHDM/RHPAM 7.11. I would suggest to use "dmn-drools-springboot-metrics-service" instead. Please change values of "app" and "name" accordingly.

@r00ta
Can you please also review and confirm.

Reviewer 2: Yes, it would be better to use `dmn-drools-springboot-metrics-service` or `dmn-drools-quarkus-metrics-service`

Author: Done

[id="con-rhpam-kogito-operator-with-prometheus-and-grafana_{context}"]
= Red Hat Process Automation Manager {KOGITO} Operator interaction with Prometheus and Grafana

> {KOGITO} in {PRODUCT} provides a `monitoring-prometheus-addon` add-on that enables Prometheus metrics monitoring for {KOGITO} microservices and generates Grafana dashboards that consume the default metrics exported by the add-on. The RHPAM {KOGITO} Operator uses the  https://github.com/coreos/prometheus-operator[Prometheus Operator] to expose the metrics from your project for Prometheus to scrape. Due to this dependency, the Prometheus Operator must be installed in the same namespace as your project.

Reviewer 3: ```suggestion
{KOGITO} in {PRODUCT} provides a `monitoring-prometheus-addon` add-on that enables Prometheus metrics monitoring for {KOGITO} microservices and generates Grafana dashboards that consume the default metrics exported by the add-on. The RHPAM {KOGITO} Operator uses the https://github.com/coreos/prometheus-operator[Prometheus Operator] to expose the metrics from your project for Prometheus to scrape. Due to this dependency, you must install the Prometheus Operator in the same namespace as your project.
```
Passive voice.

If you want to enable the Prometheus metrics monitoring for your {KOGITO} microservices, add the following dependency to the `pom.xml` file in your project, depending on the framework you are using:

.Dependency for Prometheus Quarkus add-on
[source,xml]
----
<dependency>
      <groupId>org.kie.kogito</groupId>
      <artifactId>monitoring-prometheus-quarkus-addon</artifactId>
</dependency>
----

.Dependency for Prometheus Spring Boot add-on
[source,xml]
----
<dependency>
      <groupId>org.kie.kogito</groupId>
      <artifactId>monitoring-prometheus-springboot-addon</artifactId>
</dependency>
----

> When you deploy a {KOGITO} microservice that uses the `monitoring-prometheus-addon` add-on and the Prometheus Operator is installed, the Red Hat Process Automation Manager {KOGITO} Operator creates a `ServiceMonitor` custom resource to expose the metrics for Prometheus, as shown in the following example:

Reviewer 3: ```suggestion
When you deploy a {KOGITO} microservice that uses the `monitoring-prometheus-addon` add-on, and you have installed the Prometheus Operator, the Red Hat Process Automation Manager {KOGITO} Operator creates a `ServiceMonitor` custom resource to expose the metrics for Prometheus, as shown in the following example:
```
Passive voice.

.Example `ServiceMonitor` resource for Prometheus
[source,yaml]
----
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  labels:
    app: onboarding-service
  name: onboarding-service
  namespace: kogito
spec:
  endpoints:
  - path: /metrics
    targetPort: 8080
    scheme: http
  namespaceSelector:
    matchNames:
    - kogito
  selector:
    matchLabels:
      app: onboarding-service
----

You must manually configure your `Prometheus` custom resource that is managed by the Prometheus Operator to select the `ServiceMonitor` resource:

.Example `Prometheus` resource
[source,yaml]
----
apiVersion: monitoring.coreos.com/v1
kind: Prometheus
metadata:
  name: prometheus
spec:
  serviceAccountName: prometheus
  serviceMonitorSelector:
    matchLabels:
      app: dmn-drools-quarkus-metrics-service
----

> After you configure your Prometheus resource with the `ServiceMonitor` resource, you can see the endpoint being scraped by Prometheus in the *Targets* page in the Prometheus web console. The metrics exposed by the {PRODUCT} service appear in the *Graph* view.

Reviewer 3: ```suggestion
After you configure your Prometheus resource with the `ServiceMonitor` resource, you can see the endpoint scraped by Prometheus in the *Targets* page in the Prometheus web console. The metrics exposed by the {PRODUCT} service appear in the *Graph* view.
```

"being" is redundant here.
